core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -493,6 +523,15 @@ class DAGScheduler(
     }
 
     assert(partitions.size > 0)
+
+    /**
+     * Makes sure that getPartitions occurs before
```

Comment:
This should use the `//` commenting style, not `/**` (since that's Scaladoc)`.  I'd also include a reference to SPARK-4961 so that future readers know where to go for more context.